analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -48,9 +50,14 @@ public RoslynSymbolicExecution(ControlFlowGraph cfg, SymbolicCheck[] checks)
         public void Execute()
         {
             // ToDo: Forbid running twice
+            var steps = 0;
             queue.Enqueue(new ExplodedNode(cfg.EntryBlock, ProgramState.Empty));
             while (queue.Any())
```

Comment:
Could we move the condition here `while (queue.Any() && steps++ <= MaxStepCount)`? I find it easier to understand.

Reply:
Unfortunately no, because we need to avoid `NotifyExecutionCompleted` call.